### ISSUE TITLE
or inside match statement fails to build under test

### ISSUE THREAD

**Issue by djoodle** [vlang/v#3312]
**V version:** V 0.1.24 51222ba
**OS:** 
```Distributor ID: Ubuntu
Description:    Ubuntu 19.04
Release:        19.04
Codename:       disco
```
**What did you do?**
Running a test on a method that contains a match with an embeded `or { .. }`. This code reproduces:

```v
module myexample
struct Example {
	mut:
		trigger bool
}

pub fn (ex Example) breaks() i64 {
	return match ex.trigger {
		true { i64(1) }
		false { testopt() or { i64(0) } }
		else { i64(0) }
	}
}

fn testopt() ?i64 {
	return i64(10)
}
```

and test:

```v
import (
	vtest
	myexample
)

fn test_nout() {
	assert vtest.assert_true(true)
}
```

**What did you expect to see?**
Code should compile and test should pass
 
**What did you see instead?**
```shell
gitpod /workspace/gjson $ v test gjson_test.v
Testing...
FAIL   246 ms [1/1] gjson_test.v
`/workspace/gjson/gjson_test.v`
 (
warning: gjson_test.v:61:5: the following imports were never used: 
 * gjson
/tmp/v/gjson_test.tmp.c: In function ‘gjson__Example_breaks’:
/tmp/v/gjson_test.tmp.c:3393:71: error: expected ‘)’ before ‘tmp2’
  ( ( tmp1 ==  1 ) ? ( ((i64)( 1 ) ) ) : ( ( tmp1 ==  0 ) ? (Option_i64...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen. 
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
)
--------------------------------------------------------------------------------------------------------------
     246 ms <=== total time spent running V _test.v files
                 ok, fail, total =     0,     1,     1
```

**Comment by djoodle** [comment]
Don't need to be running a test to reproduce, looks like even ininteractive mode any or within a match statement seems to through problems

**Comment by Ivo-Balbaert** [comment]
Indeed, this code does not compile:

struct Example {
	mut:
		trigger bool
}

fn (ex Example) breaks() i64 {
	return match ex.trigger {
		true { i64(1) }
		false { testopt() or { i64(0) } }
		else { i64(0) }
	}
}

fn testopt() ?int { return 6}

with error:
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c: In function 'Example_breaks':
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c:7853:13: error: expected ')' before '_t2'
   Option_int _t2 = testopt();
             ^~~~
             )
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c:7852:30: note: to match this '('
  ) : (ex.trigger == false) ? (
                              ^
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c:7859:4: error: expected expression before ':' token
  ) : (
    ^
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c: In function '_vinit':
...
==================
(Use `v -cg` to print the entire error message)

**Comment by spaceface777** [comment]
Closed in favor of #6395 which contains the same issue and is more up-to-date.
